corehq/apps/integration/payments/views.py

```diff
@@ -119,10 +119,9 @@ def kyc_config(self):
     def _get_user_or_case_ids(object_list):
         user_or_case_ids = []
         for commcare_payment_case_details in object_list:
-            for case_property_detail in commcare_payment_case_details['case_properties']:
-                if case_property_detail['key'] == PaymentProperties.USER_OR_CASE_ID:
-                    if case_property_detail['value']:
-                        user_or_case_ids.append(case_property_detail['value'])
+            case = wrap_case_search_hit(commcare_payment_case_details)
+            if case_prop := case.get_case_property(PaymentProperties.USER_OR_CASE_ID):
+                user_or_case_ids.append(case_prop)
```

Comment:
:sweat_smile:

Reply:
oh ya, that sounds weird, I will fix that.

Reply:
Fixed here, https://github.com/dimagi/commcare-hq/pull/36147/commits/ff2e60297208b34a824ae3a7b39a1b793e6d0c2f